compiler error/missing header

Looks like a missing header, google suggests this is defined in an NVIDIA library
```
smic:/work/rblas/RandBLAS/build$make -j install
[ 14%] Building CXX object src/CMakeFiles/RandBLAS.dir/sjlts.cc.o
[ 28%] Building CXX object src/CMakeFiles/RandBLAS.dir/util.cc.o
[ 42%] Building CXX object src/CMakeFiles/RandBLAS.dir/dense_op.cc.o
In file included from /work/rblas/RandBLAS/include/RandBLAS/dense_op.hh:19,
                 from /work/rblas/RandBLAS/src/dense_op.cc:1:
/work/rblas/random123-install/include/Random123/boxmuller.hpp: In function ‘r123::float2 r123::boxmuller(uint32_t, uint32_t)’:
/work/rblas/random123-install/include/Random123/boxmuller.hpp:116:5: error: ‘sincospif’ was not declared in this scope; did you mean ‘sincosf’?
  116 |     sincospif(uneg11<float>(u0), &f.x, &f.y);
      |     ^~~~~~~~~
      |     sincosf
/work/rblas/random123-install/include/Random123/boxmuller.hpp: In function ‘r123::double2 r123::boxmuller(uint64_t, uint64_t)’:
/work/rblas/random123-install/include/Random123/boxmuller.hpp:131:5: error: ‘sincospi’ was not declared in this scope; did you mean ‘sincosl’?
  131 |     sincospi(uneg11<double>(u0), &f.x, &f.y);
      |     ^~~~~~~~
      |     sincosl
make[2]: *** [src/CMakeFiles/RandBLAS.dir/build.make:104: src/CMakeFiles/RandBLAS.dir/dense_op.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:116: src/CMakeFiles/RandBLAS.dir/all] Error 2
make: *** [Makefile:146: all] Error 2

```

--- a/include/RandBLAS/dense_op.hh
+++ b/include/RandBLAS/dense_op.hh
@@ -15,10 +15,6 @@
 #include <math.h>
 #include <typeinfo>
 
-#include <Random123/philox.h>
-#include <Random123/boxmuller.hpp>
-#include <Random123/uniform.hpp>
-
 namespace RandBLAS::dense_op {
 
 template <typename T>

--- a/src/dense_op.cc
+++ b/src/dense_op.cc
@@ -7,6 +7,20 @@
 #include <math.h>
 #include <typeinfo>
 
+// The following two functions are part of NVIDIA device side math library.
+// Random123 relies on them in both host and device sources.  We can work
+// around this by defining them when not compiling device code.
+#if !defined(__CUDACC__)
+static inline void sincospif(float x, float *s, float *c) {
+    const float PIf = 3.1415926535897932f;
+    sincosf(PIf*x, s, c);
+}
+
+static inline void sincospi(double x, double *s, double *c) {
+    const double PI = 3.1415926535897932;
+    sincos(PI*x, s, c);
+}
+#endif
 #include <Random123/philox.h>
 #include <Random123/boxmuller.hpp>
 #include <Random123/uniform.hpp>

